File: algorithms/Minimum Swaps to Make Strings Equal/Minimum Swaps to Make Strings Equal.cpp
class Solution {
public:
    int minimumSwap(string s1, string s2) {
        int n = s1.size();
        int cnt1=0,cnt2=0,i=0;
        
        while(i<n){
            int x = s1[i];
            int y = s2[i++];
            if(x=='x' and y=='y') cnt1++;
            if(x=='y' and y=='x') cnt2++;
        }
        
        if((cnt1+cnt2)%2==1) return -1;
        
        return (cnt1/2) + (cnt2/2) + (cnt1%2 + cnt2%2);
    }
};

File: scripts/algorithms/M/Minimum Swaps to Make Strings Equal/Minimum Swaps to Make Strings Equal.py
class Solution:
    def minimumSwap(self, s1: str, s2: str) -> int:
        h = defaultdict(int)
        count = 0    # variable to keep track of the number of mismatches; it is impossible to make strings equal if count is odd
        for i in range(len(s1)):
            if s1[i] != s2[i]:
                count += 1
                h[s1[i]] += 1
        if count % 2 != 0:     
            return -1
        res, a, b = 0, h['x'], h['y']
        res += a // 2 + b // 2
        if a % 2 == 0:
            return res
        return res + 2